Library/Homebrew/cmd/install.rb

```diff
@@ -278,48 +278,59 @@ def install
     # formula was found, but there's a problem with its implementation).
     $stderr.puts e.backtrace if Homebrew::EnvConfig.developer?
     ofail e.message
-  rescue FormulaOrCaskUnavailableError => e
-    if e.name == "updog"
+  rescue FormulaOrCaskUnavailableError, Cask::CaskUnavailableError => e
+    # formula name or cask token
+    name = e.try(:name) || e.token
+
+    if name == "updog"
       ofail "What's updog?"
       return
     end
 
     opoo e
-    ohai "Searching for similarly named formulae..."
-    formulae_search_results = search_formulae(e.name)
-    case formulae_search_results.length
-    when 0
-      ofail "No similarly named formulae found."
-    when 1
-      puts "This similarly named formula was found:"
-      puts formulae_search_results
-      puts "To install it, run:\n  brew install #{formulae_search_results.first}"
-    else
-      puts "These similarly named formulae were found:"
-      puts Formatter.columns(formulae_search_results)
-      puts "To install one of them, run (for example):\n  brew install #{formulae_search_results.first}"
-    end
 
-    if (reason = MissingFormula.reason(e.name))
+    reason = MissingFormula.reason(name, silent: true)
+    if !args.cask? && reason
       $stderr.puts reason
       return
     end
```

Comment:
I don't see much reason to continue if a missing formula is found assuming they weren't looking for a cask.